subprojects/core/src/main/java/org/gradle/caching/internal/tasks/DefaultTaskOutputCachingBuildCacheKeyBuilder.java

```diff
@@ -97,18 +97,19 @@ public void appendTaskActionImplementations(Collection<ImplementationSnapshot> t
     }
 
     @Override
-    public void appendInputPropertyHash(String propertyName, @Nullable HashCode hashCode) {
+    public void appendInputPropertyHash(String propertyName, HashCode hashCode) {
         hasher.putString(propertyName);
-        if (hashCode != null) {
-            hasher.putHash(hashCode);
-            inputHashes.put(propertyName, hashCode);
-            LOGGER.info("Appending inputPropertyHash for '{}' to build cache key: {}", propertyName, hashCode);
-        } else {
-            valid = false;
-            // Add a random hash to make sure it shows up as changed in the build scan
-            inputHashes.put(propertyName, HashCode.fromInt(new Random().nextInt()));
-            LOGGER.info("Could not append invalid inputPropertyHash for '{}' to build cache key.", propertyName);
-        }
+        hasher.putHash(hashCode);
+        inputHashes.put(propertyName, hashCode);
+        LOGGER.info("Appending inputPropertyHash for '{}' to build cache key: {}", propertyName, hashCode);
+    }
+
+    @Override
+    public void inputPropertyLoadedByUnknownClassloader(String propertyName) {
```

Comment:
Let's use `ClassLoader` instead of `Classloader` here (and everywhere).

Reply:
done.